### PR TITLE
Use single dash for single letter command line options in QIR driver generator

### PR DESCRIPTION
**Pull request by samarsha**
This fixes an issue when using the full state simulator in Azure, where the generated entry point expects `-x` but it is invoked with the argument string `--x`.

### CODE REVIEW

File: src/Qir/Tools/Driver/QirCppDriverGenerator.cs [hunk 1]
```diff
@@ -33,20 +33,15 @@ public async Task GenerateAsync(EntryPointOperation entryPointOperation, Stream
 
         public string GetCommandLineArguments(ExecutionInformation executionInformation)
         {
-            // Sort arguments by position.
-            var sortedArguments = executionInformation.EntryPoint.Parameters.OrderBy(arg => arg.Position);
-            var argumentBuilder = new StringBuilder();
-            foreach (var arg in sortedArguments)
+            string Argument(Parameter param)
```
Comment (idavis) [inline src/Qir/Tools/Driver/QirCppDriverGenerator.cs:36]: nit: naming to something more descriptive such as `{Map/Convert}ToArgument` or something similar.

File: src/Qir/Tools/Driver/QirCppDriverGenerator.cs [hunk 2]
```diff
@@ -33,20 +33,15 @@ public async Task GenerateAsync(EntryPointOperation entryPointOperation, Stream
...
             {
-                if (argumentBuilder.Length != 0)
-                {
-                    argumentBuilder.Append(' ');
-                }
-
-                argumentBuilder.Append($"--{arg.Name}").Append(' ').Append(GetArgumentValueString(arg, executionInformation.ArgumentValues[arg.Name]));
+                var prefix = param.Name.Length > 1 ? "--" : "-";
```
Comment (idavis) [inline src/Qir/Tools/Driver/QirCppDriverGenerator.cs:38]: Is `param.Name.Length == 0` possible?

Comment (samarsha) [reply src/Qir/Tools/Driver/QirCppDriverGenerator.cs:38]: No. (I wish a type like `NonEmptyString` was more common.)

File: src/Qir/Tools/Driver/QirCppDriverGenerator.cs (hunk 1 above)
Comment (samarsha) [reply src/Qir/Tools/Driver/QirCppDriverGenerator.cs:36]: Hmm, I'm not sure that that's more descriptive... a function by definition converts its input to its output, so the name `Argument` should be enough to deduce that it converts a parameter to a string argument.

Comment (samarsha) [reply src/Qir/Tools/Driver/QirCppDriverGenerator.cs]: I renamed it to `ToArgument`.

**Review by idavis — APPROVED** [review]

**Review by cesarzc — APPROVED** [review]

**Pull request merged**